packages/nextjs/src/app-router/server/ClerkProvider.tsx

```diff
@@ -80,8 +79,16 @@ export async function ClerkProvider(
     </ClientClerkProvider>
   );
 
-  const runningWithClaimedKeys = propsWithEnvs.publishableKey === safeParseClerkFile()?.publishableKey;
-  const shouldRunAsKeyless = (!propsWithEnvs.publishableKey || runningWithClaimedKeys) && canUseKeyless;
+  let [shouldRunAsKeyless, runningWithClaimedKeys] = [false, false];
+  if (canUseKeyless) {
+    const locallyStorePublishableKey = await import('../../server/keyless-node.js')
+      .then(mod => mod.safeParseClerkFile()?.publishableKey)
+      .catch(() => undefined);
```

Comment:
[minor] For readability

```suggestion
let locallyStorePublishableKey;

try {
  const mod = await import('../../server/keyless-node.js');
  locallyStorePublishableKey = mod.safeParseClerkFile()?.publishableKey;
} catch {
  locallyStorePublishableKey = undefined;
}
```

Reply:
tbh i prefer the current one for readability